src/main.zig

```diff
@@ -2374,48 +2388,62 @@ fn buildOutputType(
                         fatal("unable to parse minor image version '{s}': {s}", .{ minor, @errorName(err) });
                     };
                     have_version = true;
-                } else if (mem.eql(u8, arg, "-e") or mem.eql(u8, arg, "--entry")) {
+                } else if (mem.eql(u8, arg, "-e") or mem.eql(u8, arg, "--entry") or mem.eql(u8, arg, "-entry")) {
                     entry = .{ .named = linker_args_it.nextOrFatal() };
-                } else if (mem.eql(u8, arg, "-u")) {
+                } else if (mem.eql(u8, arg, "-u") or mem.eql(u8, arg, "--undefined")) {
                     try force_undefined_symbols.put(arena, linker_args_it.nextOrFatal(), {});
-                } else if (mem.eql(u8, arg, "--stack") or mem.eql(u8, arg, "-stack_size")) {
+                } else if (mem.eql(u8, arg, "--stack") or mem.eql(u8, arg, "-stack") or mem.eql(u8, arg, "-stack_size")) {
                     stack_size = parseStackSize(linker_args_it.nextOrFatal());
-                } else if (mem.eql(u8, arg, "--image-base")) {
+                } else if (mem.eql(u8, arg, "--image-base") or mem.eql(u8, arg, "-image-base")) {
                     image_base = parseImageBase(linker_args_it.nextOrFatal());
-                } else if (mem.eql(u8, arg, "-T") or mem.eql(u8, arg, "--script")) {
+                } else if (mem.eql(u8, arg, "--enable-auto-image-base") or
+                    mem.eql(u8, arg, "-enable-auto-image-base") or
+                    mem.eql(u8, arg, "--disable-auto-image-base") or
+                    mem.eql(u8, arg, "-disable-auto-image-base"))
+                {
+                    // These are ignored by LLD, but are used by Libtool for MinGW triples.
```

Comment:
I don't think zig should recognize options and then do nothing.

What LLD does is not really relevant here. The relevant question is, does the zig project support this functionality or not?

If the user's request cannot be satisfied, it must be an error. If the user's request is satisfied regardless of this option, it should be a warning that the option is superfluous. If the user's request is ambiguous, the flag should not be supported.

Reply:
`--enable-auto-image-base` is a flag that binutils added 24 years ago for MinGW. It does a hash of the file and uses that as part of the image base value. I assume the idea was to avoid DLLs needing to be relocated when loaded. This is practically irrelevant today as all PEs produced since Windows Vista have ASLR enabled by default anyway, and Windows 10+ has Mandatory ASLR on by default, so it doesn't even care what the PE file wants.

We *could* implement the actual functionality behind the flag, but I'm fairly certain nobody actually cares about that anymore. The only reason the flag is remotely relevant is because it's hardcoded in Libtool, so if we don't at least accept and ignore it, it's not possible to cross-compile an Autotools project for MinGW with Zig.

Reply:
@andrewrk given this explanation, would you consider it acceptable to recognize and ignore just `--(enable,disable)-auto-image-base` as a special case?

I don't really care as much about the rest of the stuff I added in this PR, and agree that it would be nice to use LLD's TableGen files like we do for Clang. But usage of `--enable-auto-image-base` in particular is hardcoded in Libtool when targeting MinGW, and that makes `zig cc` unusable for that use case. Were it *any* other project, I would just ask them to stop using that flag (since it's pointless linker churn nowadays due to ASLR), but given the kinds of archaic systems they insist on maintaining support for in Autotools, I don't think there's any chance whatsoever of that happening. Just ignoring the flag unblocks a real use case, and [LLD having ignored it for ~7 years](https://github.com/llvm/llvm-project/commit/894dbbe8eb0e3d8fc7d8746a76d2380f0c2b6c0f#diff-bab81636d9cc1598477f8540cbe9bfaa72d97cde2ebc22cec4e218e1cd91c439R26) would indicate that no one building modern software actually cares about the effect it has in GNU `ld`.